### PR TITLE
[#9382] Migrate features to session result page

### PR DESCRIPTION
**Pull request by niqiukun**
Part of #9382 

**Outline of Solution**

Migrate the following features to session result page:

- **Add `Download Results` button**

The download configuration changes according to the settings on the page. Specifically, it can be configured to show/not show statistics and indicate/not indicate missing responses.

With both options ticked:
image

With both options unticked:
image

- **Add `Print View` button**

Expand all tabs, wait for one second for data loading to finish and open the printing dialog.

image

- **Add `Expand/Collapse All` button**

Expand/collapse all tabs (not including inner tabs).

Before expand:
image

After expand:
image

- **Add sortable table headers**

Make responses sortable by giver name, giver team, recipient name and recipient team for per-question response view.

### CODE REVIEW

**Comment by niqiukun** [comment]
@wkurniawan07 Need your review here since there are additions to UI.

**Comment by niqiukun** [comment]
@xpdavid Do you have any suggestions on UI? I am not very sure about the best way to indicate that the additional settings have an effect on the csv download (right now I use a tooltip in the screenshot above).

File: src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.ts [hunk 1]
```diff
@@ -67,4 +74,35 @@ export class PerQuestionViewResponsesComponent implements OnInit, OnChanges {
     this.responsesToShow = responsesToShow;
   }
 
+  sortResponses(by: SortBy): void {
```
Comment (xpdavid) [inline src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.ts:77]: I think there is a sort.serviece integrated recently.

Comment (niqiukun) [reply src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.ts:77]: I followed the implementation similar to `InstructorHomePageComponent` and `StudentListInfoTableComponent`. I don't think these have been refactored into sort services?

**Pull request merged**